libs/providers/rocketflag/src/lib/rocketflag-provider.ts

```diff
@@ -0,0 +1,141 @@
+import type { EvaluationContext, Provider, JsonValue, ResolutionDetails, Logger } from '@openfeature/web-sdk';
+import { ErrorCode, ProviderEvents, StandardResolutionReasons } from '@openfeature/web-sdk';
+import { EventEmitter } from 'events';
+
+export interface UserContext {
+  cohort?: string;
+}
+
+export interface FlagStatus {
+  enabled: boolean;
+}
+
+interface RocketFlagClient {
+  getFlag(flagKey: string, userContext: UserContext): Promise<FlagStatus>;
+}
+
+/**
+ * A helper function to fetch a flag, update the cache, and notify OpenFeature of changes.
+ * It's defined within the factory's scope to access the client, cache, emitter, and logger.
+ */
+const fetchFlagAndUpdateCache = (
+  // Parameters required for the fetch operation
+  flagKey: string,
+  defaultValue: boolean,
+  context: EvaluationContext,
+  cacheKey: string,
+  // State managed by the factory's closure
+  client: RocketFlagClient,
+  cache: Map<string, ResolutionDetails<boolean>>,
+  emitter: EventEmitter,
+  logger?: Logger,
+) => {
+  const userContext: UserContext = {};
+  const { targetingKey } = context;
+
+  if (targetingKey && typeof targetingKey === 'string' && targetingKey !== '') {
+    userContext.cohort = targetingKey;
+  }
+
+  client
+    .getFlag(flagKey, userContext)
+    .then((flagStatus) => {
+      const details: ResolutionDetails<boolean> = {
+        value: flagStatus.enabled,
+        reason: userContext.cohort ? StandardResolutionReasons.TARGETING_MATCH : StandardResolutionReasons.DEFAULT,
+      };
+      cache.set(cacheKey, details);
+      emitter.emit(ProviderEvents.ConfigurationChanged, { flagsChanged: [flagKey] });
+      logger?.debug(`Successfully fetched flag: ${flagKey}`);
+    })
+    .catch((error: unknown) => {
+      const err = error instanceof Error ? error : new Error(String(error));
+      const details: ResolutionDetails<boolean> = {
+        value: defaultValue,
+        reason: StandardResolutionReasons.ERROR,
+        errorCode: ErrorCode.GENERAL,
+        errorMessage: err.message,
+      };
+      cache.set(cacheKey, details);
```

Comment:
It looks like errors will be cached indefinitely. Is that okay for your use case?